FaunaDB.Client/Types/Decoder.cs

```diff
@@ -143,22 +145,66 @@ static object DecodeIntern(Value value, Type dstType)
 
                 case TypeCode.Object:
                     if (typeof(DateTimeOffset) == dstType)
-                        return new DateTimeOffset(ToDateTime(value));
+                        if (forceType == typeof(StringV) && value.GetType() == typeof(StringV))
+                            if (value != null && value != NullV.Instance)
+                                return new DateTimeOffset(Convert.ToDateTime(ToString(value)));
+                            else
+                                return null;
```

Comment:
returning `null` for a primitive type will throw an error, I think you should return the default value for the type